src/EditorFeatures/Core/Implementation/CommentSelection/CommentUncommentSelectionCommandHandler.cs

```diff
@@ -103,11 +112,19 @@ internal void ExecuteCommand(ITextView textView, ITextBuffer subjectBuffer, Oper
 
                     CollectEdits(service, textView.Selection.GetSnapshotSpansOnBuffer(subjectBuffer), textChanges, trackingSpans, operation);
 
-                    document.Project.Solution.Workspace.ApplyTextChanges(document.Id, textChanges, waitContext.CancellationToken);
+                    using (var transaction = new CaretPreservingEditTransaction(title, textView, _undoHistoryRegistry, _editorOperationsFactoryService))
+                    {
+                        document.Project.Solution.Workspace.ApplyTextChanges(document.Id, textChanges, waitContext.CancellationToken);
+                        transaction.Complete();
+                    }
 
                     if (operation == Operation.Uncomment)
                     {
-                        Format(service, subjectBuffer.CurrentSnapshot, trackingSpans, waitContext.CancellationToken);
+                        using (var transaction = new CaretPreservingEditTransaction(title, textView, _undoHistoryRegistry, _editorOperationsFactoryService))
```

Comment:
Should the title of this transaction be something else? Like "Format"? Otherwise you'll have two with the same name which seems odd.